### DISCUSSION TITLE
Generic 3D objects

### DISCUSSION THREAD

**Discussion by lodoreds** [NanoComp/meep#2254] [Ideas]
Dear Meep developers,

First of all, thank you for keeping this wonderful project alive and up to date.

Correct me if I am wrong, but in Meep it is currently not possible to instantiate a generic 3D object with arbitrary vertices.
While the prism with slanted sidewalls can be extremely useful (especially for integrated optics), it can become quite cumbersome to draw complex geometries.
Personally, I find Lumerical's planar solid (https://optics.ansys.com/hc/en-us/articles/360034901573-Planar-solid-Simulation-Object) a rather handy solution.

Have you ever felt the need for such an object? Do you think it could be added in future versions?

Cheers,
Lodovico

**Comment by stevengj** [comment]
No, we don't support arbitrary polyhedra right now. 

Of course, you can always use a `material_function` to define `ε(x,y,z)` to be any function you want, but I agree that's not quite as convenient.

If we were to support anything it seems more general to support the output of some arbitrary modeling software like Blender or Gmsh, i.e. an arbitrary triangulated surface.

This is already implemented in SCUFF-EM (https://github.com/HomerReid/scuff-em/blob/9c6d0cb7695463af803dee8d04cdae939740cdcc/libs/libscuff/PointInObject.cc), so in principle we could grab some of the code.

**Reply by lodoreds** [reply]
Thank you for the kind reply.
I had a look at the code and some examples. It is indeed a very flexible solution and I'll definitely give it a try.
Best,
Lodovico
